### PR TITLE
Implement autowiring for all services & grid types

### PR DESCRIPTION
**Pull request by fsoedjede**
Some code have been taken from here https://github.com/APY/APYDataGridBundle/tree/service_autowire_definition

### CODE REVIEW

**Comment by jmontoyaa** [comment]
Some other related issues:

https://github.com/APY/APYDataGridBundle/issues/1033
https://github.com/APY/APYDataGridBundle/issues/1030

**Comment by fsoedjede** [comment]
@romainguerrero as you were give rights to merge, can you please review this PR and merge?

File: APYDataGridBundle.php [hunk 1]
```diff
@@ -23,7 +23,6 @@ public function build(ContainerBuilder $container)
     {
         parent::build($container);
 
-        $container->addCompilerPass(new GridExtensionPass());
```
Comment (npotier) [inline APYDataGridBundle.php:26]: I'm having an issue with the removal of this compilation pass : 

as a result, the `addColumnExtension` of the `Grid` class is no longer called during the compilation. Therefore the columns extensions are not loaded, resulting to an error of type : 

> No suitable Column Extension found for column type: number

File: Resources/config/columns.xml [hunk 2]
```diff
@@ -19,48 +19,49 @@
     </parameters>
 
     <services>
-        <service id="grid.column.text" class="%grid.column.text.class%" public="false">
-            <tag name="grid.column.extension" />
```
Comment (npotier) [inline Resources/config/columns.xml:23]: Can you please explain why did you decided to change the tag name from `grid.column.extension` to `apy_grid.column` ?

File: Resources/config/services.xml [hunk 3]
```diff
@@ -20,7 +20,7 @@
             </call>
         </service>
 
-        <service id="grid" class="%grid.class%" shared="false" public="true">
+        <service id="%grid.class%" class="%grid.class%" shared="false" public="true">
```
Comment (npotier) [inline Resources/config/services.xml:23]: Shouldn't the service id be called `grid` instead of `%grid.class%` ?

File: APYDataGridBundle.php (hunk 1 above)
Comment (npotier) [inline APYDataGridBundle.php:26]: I'm having an issue with the removal of this compilation pass : 

as a result, the `addColumnExtension` of the `Grid` class is no longer called during the compilation. Therefore the columns extensions are not loaded, resulting to an error of type : 

> No suitable Column Extension found for column type: number

File: Resources/config/legacy_aliases.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="grid.twig_extension" alias="%grid.twig_extension.class%"/>
+        <service id="grid" alias="%grid.class%" public="true"/>
+        <service id="grid.manager" alias="%grid.manager.class%"/>
```
Comment (npotier) [inline Resources/config/legacy_aliases.xml:10]: I think this service should be public

**Comment by npotier** [comment]
Hello, I'm closing this PR. Feel free to reopen if necessary

**Pull request closed without merging**